### ISSUE TITLE
v-for fails to compile with noUncheckedIndexedAccess: true

### ISSUE THREAD

**Issue by micgro42** [johnsoncodehk/vue-tsc#8]
If the Typescript 4.1 option [`noUncheckedIndexedAccess`](https://devblogs.microsoft.com/typescript/announcing-typescript-4-1-beta/#no-unchecked-indexed-access) (microsoft/typescript#39560) is set to true, then rendering components with `v-for` fails with `error TS2532: Object is possibly 'undefined'.` in code that looks like the following:

```
		<HelloWorld
			v-for="item in manyItems"
			:msg="item.msg"
		/>
```

I have created a minimal failing example: https://github.com/micgro42/vue-tsc-mfe-noUncheckedIndexAccess

Somewhere internally, vue3 or vue-tsc seems to use a classic for-loop instead of `forEach` or `for`-`of` like it should?

**Comment by johnsoncodehk** [comment]
Yes it's using `for...in` instead of `for...of`, because `for...of` does not working for some object types, see: https://github.com/johnsoncodehk/vue-tsc/issues/3, https://github.com/johnsoncodehk/volar/issues/104

**Comment by johnsoncodehk** [comment]
Fixed in 0.0.17. Now use `for...of` first, if item type is any use `for...in`.

**Comment by micgro42** [comment]
It works. Thank you :blush:!
